.github/workflows/gradle-build-pr.yml

```diff
@@ -0,0 +1,110 @@
+name: Run Gradle on PRs
+on: pull_request
+jobs:
+  gradle:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    services:
+      postgres:
+        image: postgres:9.6
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: postgres
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@v2
+      - name: "Install Java"
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - name: "Install Gradle"
+        uses: broadinstitute/gradle-command-action@v1
+      - name: "Install Google Cloud SDK"
+        uses: GoogleCloudPlatform/github-actions/setup-gcloud@0.1.1
+        with:
+          version: "279.0.0"
+          export_default_credentials: true
+      - name: "Install Vault"
+        uses: broadinstitute/setup-vault@v1
+        with:
+          version: "1.3"
+      - name: "Get Vault token"
+        env:
+          ROLE_ID: ${{ secrets.ROLE_ID }}
+          SECRET_ID: ${{ secrets.SECRET_ID }}
+          VAULT_ADDR: ${{ secrets.VAULT_ADDR }}
+          GOOGLE_APPLICATION_CREDENTIALS: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
+          GOOGLE_SA_CERT: ${{ secrets.GOOGLE_SA_CERT }}
+        run: |
+          export VAULT_TOKEN=$(curl \
+            --request POST \
+            --data '{"role_id":"'"${ROLE_ID}"'","secret_id":"'"${SECRET_ID}"'"}' \
+            ${VAULT_ADDR}/v1/auth/approle/login | jq -r .auth.client_token)
+          vault read -format=json secret/dsde/datarepo/dev/sa-key.json | \
+            jq .data > $GOOGLE_APPLICATION_CREDENTIALS
+          jq -r .private_key $GOOGLE_APPLICATION_CREDENTIALS > $GOOGLE_SA_CERT
+          chmod 600 $GOOGLE_SA_CERT
+      - name: "Configure Google Cloud SDK"
+        env:
+          INTEGRATION_ZONE: us-central1
+          INTEGRATION_PROJECT: broad-jade-integration
+          CLUSTER_NAME: integration-master
+          GOOGLE_APPLICATION_CREDENTIALS: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
+        run: |
+          gcloud auth activate-service-account --key-file $GOOGLE_APPLICATION_CREDENTIALS
+
+          gcloud config set compute/zone ${INTEGRATION_ZONE}
+          gcloud config set project ${INTEGRATION_PROJECT}
+
+          gcloud auth configure-docker
+          gcloud components install kubectl
+
+          CUR_IPS=$(gcloud container clusters describe ${CLUSTER_NAME} --format json | \
+            jq -r '[.masterAuthorizedNetworksConfig.cidrBlocks[] | .cidrBlock]')
+          RUNNER_IP=$(curl https://api.ipify.org/?format=text | xargs printf '[ "%s/32" ]')
+          NEW_IPS=$(printf '%s\n' $CUR_IPS $RUNNER_IP | jq -s -r 'add | unique | join(",")')
```

Comment:
Could you add a comment explaining what this logic does please? I think I understand that you're adding an IP address as an authorized network to access the kubernetes master. And then you remove it in the cleanup step below. But I don't understand why that's necessary. Thanks.

Reply:
@marikomedlock , let me know if the added comments help.

Reply:
Yes, thanks.